backend/src/main/kotlin/app/ehrenamtskarte/backend/graphql/application/EakApplicationMutationController.kt

```diff
@@ -120,12 +121,11 @@ class EakApplicationMutationController(
         @Argument accessKey: String,
     ): Boolean =
         transaction {
-            try {
-                ApplicationEntity.find { Applications.accessKey eq accessKey }.single().status = Status.Withdrawn
-                true
-            } catch (e: IllegalArgumentException) {
-                false
-            }
+            val application = ApplicationEntity.find { Applications.accessKey eq accessKey }
+                .singleOrNull() ?: throw InvalidLinkException()
+            if (application.status != Status.Pending) throw InvalidApplicationStatusException()
+            application.status = Status.Withdrawn
+            true
```

Comment:
🔧 
- I strongly suggest to always use curly braces (`{}`) and an indented block for `if` constructs, unless it is used as an expression.
- Also, when in a tail position, it is easier to see when (or when not) the code following the if clause is executed:

```suggestion
            val application = ApplicationEntity.find { Applications.accessKey eq accessKey }
                .singleOrNull() ?: throw InvalidLinkException()
         
            if (application.status != Status.Pending) {
                throw InvalidApplicationStatusException()
            } else {
                application.status = Status.Withdrawn
                true
            }
```

❓ The return type here does not carry any useful information?

Reply:
You are right, mutations need to return some data.